src/components/AllProducts/index.js

```diff
@@ -0,0 +1,50 @@
+import React from "react";
+import { withRouter,Link, Redirect } from "react-router-dom";
+import { connect } from 'react-redux';
+import {Button,Container, Col, Row,Image,Card, Spinner} from 'react-bootstrap';
+import Footer from '../Footer';
+
+class AllProducts extends React.Component {
+    constructor(props) {
+      super(props);
+    }
+render()
+{
+    const {products}=this.props;
+    if(products.length===0){
```

Comment:
@Wajahat-Khan This check is not valid, if server returns no products then spinner will be shown even request is succeeded, so first you need to dispatch and action when fetching the products and decide to show spinner on isFetching state.

Reply:
Actually, When products will not be returned (in case of error), Error pop-up will be shown, I have catered this in Failure of Products Action.